jerry-core/lit/lit-strings.c

```diff
@@ -31,89 +31,53 @@ lit_is_valid_utf8_string (const lit_utf8_byte_t *utf8_buf_p, /**< utf-8 string *
                           lit_utf8_size_t buf_size, /**< string size */
                           bool is_strict) /**< true if surrogate pairs are not allowed */
 {
-  lit_utf8_size_t idx = 0;
+  if (!utf8_buf_p)
```

Comment:
What is the purpose of this check? Is it a valid case?

Reply:
`TEST_ASSERT (!lit_is_valid_utf8_string (NULL, sizeof (NULL), true));` (test-strings.c:247). If the pointer is NULL the run will fail without this check.

Reply:
sizeof (NULL) is 4 or 8, this check is certainly bad